Content.Server/Engineering/EntitySystems/DisassembleOnAltVerbSystem.cs

```diff
@@ -1,69 +1,24 @@
-    public sealed class DisassembleOnAltVerbSystem : EntitySystem
-    {
-        [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
-
-        public override void Initialize()
-        {
-            base.Initialize();
-
-            SubscribeLocalEvent<DisassembleOnAltVerbComponent, GetVerbsEvent<AlternativeVerb>>(AddDisassembleVerb);
-        }
-        private void AddDisassembleVerb(EntityUid uid, DisassembleOnAltVerbComponent component, GetVerbsEvent<AlternativeVerb> args)
-        {
-            if (!args.CanInteract || !args.CanAccess || args.Hands == null)
-                return;
-
-            AlternativeVerb verb = new()
-            {
-                Act = () =>
-                {
-                    AttemptDisassemble(uid, args.User, args.Target, component);
-                },
-                Text = Loc.GetString("disassemble-system-verb-disassemble"),
-                Priority = 2
-            };
-            args.Verbs.Add(verb);
-        }
-
-        public async void AttemptDisassemble(EntityUid uid, EntityUid user, EntityUid target, DisassembleOnAltVerbComponent? component = null)
-        {
-            if (!Resolve(uid, ref component))
-                return;
-            if (string.IsNullOrEmpty(component.Prototype))
-                return;
-
-            if (component.DoAfterTime > 0 && TryGet<SharedDoAfterSystem>(out var doAfterSystem))
-            {
-                var doAfterArgs = new DoAfterArgs(EntityManager, user, component.DoAfterTime, new AwaitedDoAfterEvent(), null)
-                {
-                    BreakOnMove = true,
-                };
-                var result = await doAfterSystem.WaitDoAfter(doAfterArgs);
+namespace Content.Server.Engineering.EntitySystems;
 
-                if (result != DoAfterStatus.Finished)
-                    return;
-            }
-
-            if (component.Deleted || Deleted(uid))
-                return;
-
-            if (!TryComp(uid, out TransformComponent? transformComp))
-                return;
+public sealed class DisassembleOnAltVerbSystem : EntitySystem
+{
+    [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
 
-            var entity = EntityManager.SpawnEntity(component.Prototype, transformComp.Coordinates);
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DisassembleOnAltVerbComponent, DissembleDoAfterEvent>(OnDissembleDoAfter);
+    }
 
-            _handsSystem.TryPickup(user, entity);
+    private void OnDissembleDoAfter(Entity<DisassembleOnAltVerbComponent> entity, ref DissembleDoAfterEvent args)
+    {
+        var spawnedEnt = SpawnNextToOrDrop(entity.Comp.PrototypeToSpawn, entity.Owner);
 
-            EntityManager.DeleteEntity(uid);
-        }
+        _handsSystem.TryPickup(args.User, spawnedEnt);
+        // Only reason this is in server is because this isn't predicted.
+        EntityManager.DeleteEntity(entity.Owner);
```

Comment:
Ok, it took me a while to look this up, but appearently the way it is done elsewhere is to simply do `if (_net.IsServer)` in Shared, so do that and get rid of the whole server part. Also use `QueueDel(entity.Owner)` for deleting

Reply:
What is really odd is after doing this it looks MORE laggy! I don't understand it at all 😢
Also see my message in discord, apparently spawning something and picking it up also has warnings...

Reply:
Just updated this so I could do the Voice Mask stuff... Right now its "working" but still spitting out warnings and having issues! What I'm trying to say, DONT merge it 😆

Reply:
OK I've tried experimenting, basically any prediction makes it somehow look worse and or error out so I just made it not predicted!

The way I did it is a little odd so if you want me to just directly move it to server let me know (It does work though and its nice that it's all in one files instead of two!)